src/Gui/View3DPy.cpp

```diff
@@ -1498,6 +1506,120 @@ Py::Object View3DInventorPy::getObjectsInfo(const Py::Tuple& args)
     }
 }
 
+Py::Object View3DInventorPy::getObjectInfoRay(const Py::Tuple& args)
+{
+    PyObject* vs;
+    PyObject* vd;
+    double vsx,vsy,vsz;
+    double vdx,vdy,vdz;
+    if (PyArg_ParseTuple(args.ptr(), "O!O!", &Base::VectorPy::Type, &vs,
+                                       &Base::VectorPy::Type,  &vd)) {
+        Base::Vector3d* startvec = static_cast<Base::VectorPy*>(vs)->getVectorPtr();
+        vsx = startvec->x;
+        vsy = startvec->y;
+        vsz = startvec->z;
+        Base::Vector3d* dirvec = static_cast<Base::VectorPy*>(vd)->getVectorPtr();
+        vdx = dirvec->x;
+        vdy = dirvec->y;
+        vdz = dirvec->z;
+    }
+    else {
+        PyErr_Clear();
+        if (!PyArg_ParseTuple(args.ptr(), "dddddd", &vsx,&vsy,&vsz,&vdx,&vdy,&vdz)) {
+            throw Py::TypeError("Wrong arguments, two Vectors or six floats expected expected");
+        }
+    }
+
+    try {
+
+        // As this method could be called during a SoHandleEventAction scene
+        // graph traversal we must not use a second SoHandleEventAction as
+        // we will get Coin warnings because of multiple scene graph traversals
+        // which is regarded as error-prone.
+        SoRayPickAction action(getView3DIventorPtr()->getViewer()->getSoRenderManager()->getViewportRegion());
+        action.setRay(SbVec3f(vsx, vsy, vsz), SbVec3f(vdx, vdy, vdz));
+        action.apply(getView3DIventorPtr()->getViewer()->getSoRenderManager()->getSceneGraph());
+        SoPickedPoint *Point = action.getPickedPoint();
+
+        Py::Object ret = Py::None();
+        if (Point) {
+            Py::Dict dict;
+            SbVec3f pt = Point->getPoint();
+            dict.setItem("x", Py::Float(pt[0]));
+            dict.setItem("y", Py::Float(pt[1]));
+            dict.setItem("z", Py::Float(pt[2]));

```

Comment:
The only reason for separate floats: similar existing getObjectInfo did it that way. But since, nothing uses the new code yet (so no breaking compatibility), I agree - Vector makes more sense.